sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1776,23 +1776,29 @@ case class MakeDate(year: Expression, month: Expression, day: Expression)
   override def dataType: DataType = DateType
   override def nullable: Boolean = true
 
+  val ansiEnabled: Boolean = SQLConf.get.ansiEnabled
+
   override def nullSafeEval(year: Any, month: Any, day: Any): Any = {
     try {
       val ld = LocalDate.of(year.asInstanceOf[Int], month.asInstanceOf[Int], day.asInstanceOf[Int])
       localDateToDays(ld)
     } catch {
-      case _: java.time.DateTimeException => null
+      case e: java.time.DateTimeException =>
```

Comment:
No big deal but I suppose this could be just ...
```
  case e: java.time.DateTimeException if !ansiEnabled => null
```

Reply:
done